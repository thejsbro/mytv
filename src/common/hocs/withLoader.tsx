import React from 'react';
import 'common/styles'

interface IWithLoaderHOC {
    loading: boolean;
}

export const withLoader =
        <T, >(WrappedComonent: React.ComponentType<T>) => {
            return class WithLoaderHOC extends React.Component<T & IWithLoaderHOC> {

            renderLoading = () => (<div className='center text-large'>Loading...</div>);

            render() {
                return this.props.loading ? this.renderLoading() : <WrappedComonent {...this.props} />;
            }   
    }
}
