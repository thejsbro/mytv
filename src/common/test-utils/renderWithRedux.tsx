import '@testing-library/jest-dom/extend-expect'
import React from 'react'
import {render} from '@testing-library/react'
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import reducers from 'common/reducers'

export function renderWithRedux(
    ui: JSX.Element,
    {initialState, store = createStore(reducers, initialState)}: any = {},
  ) {
    return {
      ...render(<Provider store={store}>{ui}</Provider>),
      store,
    }
  }