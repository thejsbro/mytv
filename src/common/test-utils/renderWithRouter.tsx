import React from 'react'
import {Router} from 'react-router-dom'
import {createMemoryHistory} from 'history'
import {render} from '@testing-library/react'

export function renderWithRouter(
    ui: JSX.Element,
    {route = '/', history = createMemoryHistory({initialEntries: [route]})}: any = {},
  ) {
    return {
      ...render(<Router history={history}>{ui}</Router>),
      history,
    }
  }