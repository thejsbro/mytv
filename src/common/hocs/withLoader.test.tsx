import React from 'react';
import {withLoader} from './withLoader';
import {renderWithRedux} from 'common/test-utils/renderWithRedux';
import {render, cleanup} from '@testing-library/react'

afterEach(cleanup)

class MockApp extends React.Component<{loading: boolean}> {
    render () {
      return (
        <p>
          Test App
        </p>
      )
    }
  }

const MockWithHOC = (withLoader(MockApp))

test('can render when loading', () => {
    const { container, getByTestId } = renderWithRedux(
        <MockWithHOC loading={true} />)
        
    expect(getByTestId('loading')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
})

test('can render when loaded', () => {
    const { container, getByText } = renderWithRedux(
        <MockWithHOC loading={false} />)

    expect(getByText(/Test App/i)).toBeInTheDocument()

    expect(container).toMatchSnapshot()
})
