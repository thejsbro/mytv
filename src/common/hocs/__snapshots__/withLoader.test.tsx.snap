// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render when loaded 1`] = `
<div>
  <p>
    Test App
  </p>
</div>
`;

exports[`can render when loading 1`] = `
<div>
  <div
    class="center text-large"
    data-testid="loading"
  >
    Loading...
  </div>
</div>
`;
